library/manageiq.py

```diff
@@ -64,14 +61,86 @@
 '''
 
 
-def update_required(client, module, providers_url, provider_id, hostname, port):
-    try:
-        result = client.get(providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
-        endpoint = result['endpoints'][0]
-        return False if (endpoint['hostname'] == hostname and endpoint['port'] == int(port)) else True
-    except Exception as e:
-        msg = "Failed to get provider data. Error: %s" % e
-        module.fail_json(msg=msg)
+class ManageIQ(object):
+    """
+        ManageIQ object to execute various operations in manageiq
```

Comment:
This description doesn't say much, but I don't have concrete improvement suggestion.

nitpick: align under """ (4 spaces not 8) please